jerry-core/ecma/base/ecma-gc.c

```diff
@@ -199,6 +199,136 @@ ecma_gc_init (void)
   ecma_gc_new_objects_since_last_gc = 0;
 } /* ecma_gc_init */
 
+static void
+ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
+{
+  switch (ECMA_OBJECT_GET_PROPERTY_TYPE (property_p))
+  {
+    case ECMA_PROPERTY_TYPE_NAMEDDATA:
+    {
+      ecma_value_t value = ecma_get_named_data_property_value (property_p);
+
+      if (ecma_is_value_object (value))
+      {
+        ecma_object_t *value_obj_p = ecma_get_object_from_value (value);
+
+        ecma_gc_set_object_visited (value_obj_p, true);
+      }
+      break;
+    }
+    case ECMA_PROPERTY_TYPE_NAMEDACCESSOR:
+    {
+      ecma_object_t *getter_obj_p = ecma_get_named_accessor_property_getter (property_p);
+      ecma_object_t *setter_obj_p = ecma_get_named_accessor_property_setter (property_p);
+
+      if (getter_obj_p != NULL)
+      {
+        ecma_gc_set_object_visited (getter_obj_p, true);
+      }
+
+      if (setter_obj_p != NULL)
+      {
+        ecma_gc_set_object_visited (setter_obj_p, true);
+      }
+      break;
+    }
+    case ECMA_PROPERTY_TYPE_INTERNAL:
+    {
+      uint32_t property_value = ECMA_OBJECT_PROPERTY_VALUE (property_p)->value;
+
+      switch (ECMA_OBJECT_GET_INTERNAL_PROPERTY_TYPE (property_p))
+      {
+        case ECMA_INTERNAL_PROPERTY_NUMBER_INDEXED_ARRAY_VALUES: /* a collection of ecma values */
+        case ECMA_INTERNAL_PROPERTY_STRING_INDEXED_ARRAY_VALUES: /* a collection of ecma values */
+        {
+          JERRY_UNIMPLEMENTED ("Indexed array storage is not implemented yet.");
```

Comment:
That code is copied. I don't want to change that in this patch. I want to work on the internal properties in the next patch.